content/livesync/frontend.textile

```diff
@@ -0,0 +1,373 @@
+- @eventBufferOptions@ := used to configure the in-memory sliding-window buffer used for reordering and deduplication.
+- @optimisticEventOptions@ := is used to configure how "optimistic events":/livesync/models/models#optimistic are applied.
+- @logLevel@ := configures the log level used to control the verbosity of log output. One of @fatal@, @error@, @warn@, @info@, @debug@, or @trace@.
+- @ModelsClient@ := captures a collection of named model instances used in your application and provides methods for creating new models.
+
+The following is an example of setting @ClientOptions@ when instantiating the Models SDK:
+
+```[javascript]
+const modelsClient = new ModelsClient({
+ client,
+ logLevel,
+ syncOptions: {
+  historyPageSize,
+  messageRetentionPeriod,
+  retryStrategy,
+ },
+ eventBufferOptions: {
+  bufferMs,
+  eventOrderer,
+ },
+ optimisticEventOptions: {
+  timeout,
+ },
+});
+```
+
+h2(#create-or-retrieve). Create or retrieve a model
+
+Create a model using the @models.get()@ method on the client. If you specify a name that already exists, it will be returned.
+
+Instantiate a model using a unique channel name. A unique channel name identifies the model on the client and corresponds to the "channel":/channels name used to subscribe to state updates from the backend:
+
+```[javascript]
+const model = modelsClient.models.get({
+  name: 'Posts',
+  channelName: 'posts',
+  sync,
+  merge,
+});
+```
+
+TODO: Diagram of how the model components work together
+
+h2(#designing). Designing your Channels and Models
+
+In general, when designing your channels and models, aim for a 1:1:1 relationship between channels, models, and domain objects. Channels facilitate data partitioning and access control. For instance, in a task management app, each task can be represented by a model with its own channel, ensuring updates are specific to each task. Ably's "Token Capabilities":/auth/capabilities allow fine-grained access control.
+
+Having each model on its own channel reduces processing overhead, ensuring models only receive relevant messages. While multiple models can share a channel, it requires careful message filtering. Channels aid scalability, but finding the right balance between channels, models, and message frequency is crucial.
+
+For infrequently updated models, grouping them on the same channel may be viable, while frequently changing models benefit from individual channels. Starting with one channel per model and one model per domain object is a good baseline, adjusting as needed.
+
+h2(#create-sync). Create a Sync function
+
+The sync function instructs your model to initialize with the latest data from the backend and is necessary when creating a model. It can be any function that optionally accepts parameters and returns a promise with the latest state of your data model, along with a "@sequenceId@":#sequence-id.
```

Comment:
I think this is confusing and doesn't help me understand what the purpose of a sync function is. I think we should separate what a certain functionality offers to the customers vs what it facilitates towards the working of the library itself.

As a developer reading this, the questions I have in mind are "what is a sync function? why is it needed? what does it help me do? how does it help the models sdk  work correctly?

Reply:
It's also not clear when we say backend, if we mean their backend or our backend, for example.